hub/core/dataset/dataset.py

```diff
@@ -305,6 +308,8 @@ def create_tensor(
             **kwargs: `htype` defaults can be overridden by passing any of the compatible parameters.
                 To see all `htype`s and their correspondent arguments, check out `hub/htypes.py`.
             hidden (bool): If True, the tensor will be hidden from ds.tensors but can still be accessed via ds[tensor_name]
+            create_id_tensor (bool): If True, an associated tensor containing unique ids for each sample will be created.
```

Comment:
keep in mind that you would need to replace it with hashes of raw data if necessary later

Reply:
keeping hashes would defeat the purpose of these ids helping us figure out of conflicts, hashes could be a separate thing we store to check the data integrity.

Reply:
if not wrong, the purpose of cryptographic hashes is to figure out conflicts (not only integrity).

Reply:
the use case here is different, in git files are identified by their names, in hub samples don't have any unique name, id is essentially that. It allows us to track samples even if they were updated, for which the id needs to be a constant that doesn't change when underlying data changes.

Reply:
usually in that case you introduce immutability where the original raw data doesn't change, only references change. Also compositions of data become possible, though if you update one element in the array that is basically new hash. (unless you hash tiles which is possible).

Reply:
@davidbuniat it shouldn't be hash as updating a sample shouldnt change its id.